.github/workflows/cloudbeat-ci.yml

```diff
@@ -82,63 +85,66 @@ jobs:
           rm -rf /tmp/.buildx-cache
           mv /tmp/.buildx-cache-new /tmp/.buildx-cache
 
-  Integration_tests:
-    name: Integration Tests
+  Test_Matrix:
+    name: ${{ matrix.test-target }}-tests
+    needs: Build
     runs-on: ubuntu-latest
     timeout-minutes: 20
+    strategy:
+      matrix:
+        test-target: [pre_merge, file_system_rules, k8s_object_rules, process_api_server_rules, process_controller_manager_rules, process_etcd_rules, process_kubelet_rules, process_scheduler_rules]
+      fail-fast: false
```

Comment:
Pre merge integration tests should execute only **pre_merge** marker tests as required.
Functional tests may be executed but not as required.

Reply:
Added pre_merge & file_system rules. Only pre_merge will be set as mandatory check in gh branch rules.